packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -238,12 +246,15 @@ class RenderEditable extends RenderBox {
   /// The default value of this property is false.
   bool ignorePointer;
 
-  /// Whether text is composed.
```

Comment:
This will be a breaking change, I really regret to put in this field in read only pr. As thing get more sophisticated with texteditingcontroller, I realize there is no way for RenderEditable to know whether the text is composed or not

Reply:
SelectableText will be supporting rich text, it will always be a textspan tree in that case. I have thought about checking the exact format, ie composing will make the second child of the text span to have underline decoration, but that is not secure as well since selectabletext richtext can have any arbitrary structure.